realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -161,10 +176,12 @@ long getNativePtr() {
 
     public void beginTransaction() {
         nativeBeginTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
     }
 
     public void commitTransaction() {
         nativeCommitTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
```

Comment:
Current implementation, `DynamicRealm` on the same thread does not trigger any listener because I think that the `VersionID` of the each `SharedRealm` is independent.

If my thought is correct, schema changes by Dynamic API can be seen at the next `advance_read` or `beginTransaction()` by relevant typed Realm.